Tactility/Source/service/wifi/WifiApSettings.cpp

```
@@ -56,60 +58,70 @@ static std::string getApPropertiesFilePath(std::shared_ptr<ServicePaths> paths,
+// value that's known and identical at both encrypt time (save) and decrypt time (load).
+static bool encrypt(const std::string& ssid, const std::string& plaintext, std::string& ciphertextOutput) {
     uint8_t iv[16];
-    const auto length = ssidInput.size();
+    const auto length = plaintext.size();
     constexpr size_t chunk_size = 16;
     const auto encrypted_length = ((length / chunk_size) + (length % chunk_size ? 1 : 0)) * chunk_size;
 
+    // crypt_encrypt reads encrypted_length bytes, but plaintext.c_str() only guarantees length + 1 bytes,
+    // so pad the input into a zero-filled buffer of encrypted_length first to avoid reading past it.
+    auto* padded_plaintext = static_cast<uint8_t*>(calloc(encrypted_length, 1));
+    memcpy(padded_plaintext, plaintext.c_str(), length);
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)